src/audio/base_fw.c

```diff
@@ -323,6 +325,69 @@ static uint32_t basefw_get_ext_system_time(uint32_t *data_offset, char *data)
 	return IPC4_UNAVAILABLE;
 }
 
+static int basefw_register_kcps(bool first_block,
+				bool last_block,
+				uint32_t data_offset_or_size,
+				const char *data)
+{
+	if (!(first_block && last_block))
+		return IPC4_INVALID_REQUEST;
+
+	if (core_kcps_adjust(0, *(int *)data))
```

Comment:
This would need some inline documentation. First the cast to "int" looks not right, how can we assume the "int" type matches the data received via IPC. Should this be a fixed-width type like "int32_t"?

Second, why this is adding core 0 here, is this implicit part of the IPC4 protocol?